stripe/src/main/java/com/stripe/android/view/CardNumberEditText.java

```diff
@@ -164,7 +170,11 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             @Override
             public void afterTextChanged(Editable s) {
                 if (s.length() == mLengthMax) {
+                    boolean before = mIsCardNumberValid;
                     mIsCardNumberValid = CardUtils.isValidCardNumber(s.toString());
+                    if (!before && mIsCardNumberValid && mCardNumberCompleteListener != null) {
```

Comment:
No, the cardNumberCompleteListener is invoked by the CardInputView (which is what tells it to rotate)